docs/export-map.rst

```diff
@@ -0,0 +1,163 @@
+export-map
+==========
+
+.. dfhack-tool::
+    :summary: Export fortress map tile data to a JSON file
+    :tags: dev map
+
+WARNING - This command will cause the game to freeze for minutes depending on
+map size and options enabled.
```

Comment:
if the runtime for large maps is untenable, it probably would not be too hard to move the property scanning logic a Lua-accessible function of `probe` and just use Lua for the output formatting (which is a pain to do in C++).